esx_service/utils/auth_data.py

```diff
@@ -319,6 +337,14 @@ def connect(self):
         Raises a ConnectionFailed exception when connect fails.
 
         """
+        global has_migrate_auth_db
+
+        if os.path.isfile(self.db_path):
+            # already exist
+            if not has_migrate_auth_db and self.need_migrate_auth_db():
```

Comment:
self.need_migrate_auth_db() need to check for 'has already migrated' (what you cann  has_migrate_auth_db ) for better encapsulation